feat: active admin frontend to manage locations

bbazyk opened the pull request:
![helios active admin](https://user-images.githubusercontent.com/84336508/121254895-cb74eb00-c878-11eb-8a97-482b671dc939.gif)

ticket id: 178270028
accessible through http://localhost:5000/admin
user email: admin@example.com
password: password
Use the above tabs to navigate models and edit their parameters. Can be used to alter the displayed wifi and bathroom codes, rearrange and adjust widget screen time, or even delete a traffic cam from view.

chewnoill commented on config/initializers/active_admin.rb:
```diff
@@ -0,0 +1,335 @@
+  #
+  config.site_title = "Helios2"
+
+  # Set the link url for the title. For example, to take
```
I think we can remove all of these comments, these files are all pretty simple

chewnoill commented on db/seeds.rb:
```diff
@@ -186,3 +186,5 @@
   r.feed_format = "video"
 end.save!
+
+AdminUser.create!(email: 'admin@example.com', password: 'password', password_confirmation: 'password') if Rails.env.development?
```
Let's make sure that the seed script is **idempotent**. It should be able to be run more then once without breaking.

chewnoill commented on app/admin/events.rb:
```diff
@@ -5,7 +5,7 @@
   # Uncomment all parameters which should be permitted for assignment
   #
-  # permit_params :source, :external_id
+  permit_params :source, :external_id
```
lets squash this commit with cb3e566b8df8df287ab1890bf8855749a46c4391 and remove all of these extra comments

chewnoill commented on db/seeds.rb:
```diff
@@ -187,4 +187,4 @@
 end.save!
 
-AdminUser.create!(email: 'admin@example.com', password: 'password', password_confirmation: 'password') if Rails.env.development?
+AdminUser.create!(email: 'helios@mojotech.com', password: 'mojojojo', password_confirmation: 'mojojojo') if Rails.env.development?
```
if this is only for development, its nice to keep the username/password simpler: admin@example.com/password is pretty standard for development

chewnoill replied:
also, lets make sure the readme gets updated

tucker-weed commented on db/seeds.rb:
```diff
@@ -186,3 +186,11 @@
   r.feed_format = "video"
 end.save!
+
+if Rails.env.development?
+  AdminUser.create!(
+    email: 'helios@mojotech.com',
+    password: 'mojojojo',
+    password_confirmation: 'mojojojo'
+  )
+end
```
Should there be an else for this conditional to create an admin user for production?

bbazyk replied:
When rails boots up, it would automatically start with any file named development (if in development mode). It doesn't mean that the development stage isn't executed only once, and the difference between the environments has more to do with caching. Either way, the internet doesn't recommend any code of the format: 
if Rails.env.production?
  do_additional_work
end
so I'll try looking into something else...

cbarber reviewed (commented):
Looking good. I agree with Will that some of these feel like sub-atomic commits. I could see the case for adding deps being separate from adding functionality or squashing them all.

I noticed some usability issues where Widgets are showing either the Location#id or the Location#object_id where we could opt into showing the city name instead:
![Screenshot from 2021-06-10 17-53-05](https://user-images.githubusercontent.com/524850/121602242-0449c700-ca15-11eb-82fa-20cc68075beb.png)
![Screenshot from 2021-06-10 17-52-17](https://user-images.githubusercontent.com/524850/121602256-0875e480-ca15-11eb-8af2-a9310ee256f1.png)

tmsavage commented on db/seeds.rb:
```diff
@@ -186,3 +186,11 @@
+    email: 'admin@example.com',
+    password: 'password',
+    password_confirmation: 'password'
+  )
```
So this creates an admin user and their credentials are stored into the database?

chewnoill commented on config/initializers/devise.rb:
```diff
@@ -0,0 +1,313 @@
+
+  # When set to false, does not sign a user in automatically after their password is
+  # changed. Defaults to true, so a user is signed in automatically after changing a password.
+  # config.sign_in_after_change_password = true
```
I think we can remove all these comments also

chewnoill commented on db/seeds.rb:
```diff
@@ -186,3 +186,11 @@
   r.feed_format = "video"
 end.save!
+
+if Rails.env.development?
```
I would like to create an admin user in production also. Maybe something like this:

  AdminUser.create!(
    email: ENV['ADMIN_EMAIL']
    password: ENV['ADMIN_PASSWORD'],
    password_confirmation: ENV['ADMIN_PASSWORD']
  )

tucker-weed commented on config/routes.rb:
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  devise_for :admin_users, ActiveAdmin::Devise.config
+  ActiveAdmin.routes(self)
```
What does this do?

sandipa12 replied:
Is it to use devise to log in regular users with same user model?

fallynlogan commented on db/migrate/20210607180232_devise_create_admin_users.rb:
```diff
@@ -0,0 +1,43 @@
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      # t.integer  :sign_in_count, default: 0, null: false
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
```
Would we want to keep all of these in the code (for reference or to be used later) or should these comments be deleted?

bbazyk replied:
I just asked about it, and most blocks of comments and documentation that come with added files can be deleted for simplicity's sake. They're not that difficult to find again from the source (if say someone else on the project wanted some direction for the code they're tasked with).

fallynlogan commented on spec/models/admin_user_spec.rb:
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe AdminUser, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
What is this doing?

bbazyk replied:
This file was added by active admin, I've found other examples online and they all seem to contain code relevant for creating a more accurate login page (making sure the username contains a valid email address, rejecting those that are invalid, rejecting two accounts with the same email address, etc.)

chewnoill commented on db/seeds.rb:
```diff
@@ -186,3 +186,9 @@
   r.feed_format = "video"
 end.save!
+
+AdminUser.create!(
```
Maybe `find_or_create_by` is enough?

bbazyk commented:
Removed comments from code, added 'if statement' to seeds.rb that enables reseeding without breaking, admin user for rails production mode, and display city names instead of location IDs in the widgets model for active admin.
Ready for one final review? :) @chewnoill @cbarber @jennyckaplan

chewnoill reviewed (approved):
Nice work @bbazyk
